tester/queue.go

```diff
@@ -82,15 +83,21 @@ func (q *queue) waitConsumersInit() {
 	logger.Printf("Consumers in Queue %s", q.topic)
 	for cons := range q.groupConsumers {
 		logger.Printf("waiting for group consumer %s to be running", cons.queue.topic)
-		<-cons.state.WaitForState(running)
-		logger.Printf(" --> %s is running", cons.queue.topic)
+		select {
+		case <-cons.state.WaitForState(killed):
+			log.Printf("At least one consumer was killed. No point in waiting for it")
+			return
+		case <-cons.state.WaitForState(running):
+			logger.Printf(" --> %s is running", cons.queue.topic)
+		}
 	}
 
 	for cons := range q.simpleConsumers {
 		logger.Printf("waiting for simple consumer %s to be ready", cons.queue.topic)
 		select {
 		case <-cons.state.WaitForState(running):
 		case <-cons.state.WaitForState(stopped):
+		case <-cons.state.WaitForState(killed):
```

Comment:
a bit weird to be ready when it's killed.

Reply:
okay. any suggestions how to improve?